doc/develop/languages/rust/bindings.rst

```diff
@@ -0,0 +1,52 @@
+.. _rust_bindings:
+
+Bindings to Zephyr for Rust
+###########################
+
+Zephyr is written in C, and it's primary API is also made available to C.  It is written as a
+mixture of types, function prototypes, inline functions, and macros.
+
+Although Rust interfaces fairly easily with C at an ABI level, the compiler does not have any
+support for generating the code necessary for these bindings. In order to call C code, the types
+must be reproduced in Rust, function prototypes written in Rust, and often wrappers made for inline
+functions and macros (or this functionality replicated in Rust).
+
+This is a tedious, and error-prone process, and the end result would quickly get out of date with
+the development of Zephyr.
+
+The `bindgen`_ project seeks to address much of this issue by automatically generating bindings to C
+code for Rust.  It makes use of ``libclang`` to parse and interpret a given set of headers, and
+generate the Rust code needed to call this code.
+
+.. _bindgen: https://github.com/rust-lang/rust-bindgen
+
+Because the Zephyr headers use numerous conditional compilation macros, the bindings needed will be
+very specialized to a given board, and even a given configuration.  To do this, the file
+:file:`lib/rust/zephyr-sys/build.rs`, which ``cargo`` knows to run at build time will
+generate the bindings for the particular build of Zephyr being used by the current Rust application.
+
+These bindings will be made available in the ``zephyr-sys`` crate, as well as under ``zephyr::raw``.
+
+Using the Bindings
+******************
+
+In general, using direct bindings to C function from Rust is a bit more difficult than calling them
+from C.  Although all of these calls are considered "unsafe", and must be placed in an ``unsafe``
+block, the rust language has stricter constraints on what is allowed, even by unsafe code.  Although
+the intent of the Rust on Zephyr project is to allow full use of Zephyr, without needing to resort
```

Comment:
I don't necessarily dislike the idea of calling this "Rust on Zephyr", but probably better to say something like "... supporting the Rust language in Zephyr ..." or something.

Also, you have two consecutive sentences starting with "Although" :)